internal/admission/validation/gateway/httproute.go

```diff
@@ -97,13 +98,16 @@ func validateHTTPRouteListener(listener *gatewayapi.Listener) error {
 // validateHTTPRouteFeatures checks for features that are not supported by this
 // HTTPRoute implementation and validates that the provided object is not using
 // any of those unsupported features.
-func validateHTTPRouteFeatures(httproute *gatewayapi.HTTPRoute) error {
+func validateHTTPRouteFeatures(httproute *gatewayapi.HTTPRoute, parserFeatures parser.FeatureFlags, kongVersion semver.Version) error {
 	for _, rule := range httproute.Spec.Rules {
 		for _, match := range rule.Matches {
-			// We don't support query parameters matching rules yet
+			// We support query parameters matching rules only with Kong 3.4.1 and newer, and expression router.
 			// See: https://github.com/Kong/kubernetes-ingress-controller/issues/3679
 			if len(match.QueryParams) != 0 {
-				return fmt.Errorf("queryparam matching is not yet supported for httproute")
+				if !parserFeatures.ExpressionRoutes || kongVersion.LE(versions.QueryParameterVersionCutoff) {
```

Comment:
Cutoff version check is no longer needed since the global check
- https://github.com/Kong/kubernetes-ingress-controller/pull/4766

has been already merged

Reply:
removed the related version check.